src/ParallelAlgorithms/Amr/Criteria/TruncationError.cpp

```diff
@@ -61,7 +61,7 @@ void max_over_components(
           -PowerMonitors::relative_truncation_error(modes, modes.size() - 1));
       const double abs_truncation_error_coarsened =
           umax * rel_truncation_error_coarsened;
-      if (rel_truncation_error_coarsened <= target_rel_truncation_error and
+      if (rel_truncation_error_coarsened <= target_rel_truncation_error or
           abs_truncation_error_coarsened <= target_abs_truncation_error) {
```

Comment:
Could we recast this and the above check into something like:
```
err = abs(error)/(atol + rtol * abs(u))
```
I ask because I find that much easier to understand than the logic here, and it nicely maps onto 17.2.7-17.2.9 in Numerical Recipes. I think this does effectively that, if I understand correctly.

Up to you, just an idea on how to eliminate the logic, because I think the logic is subtle and requires more thought.